app/client/websocket/index.test.ts

```diff
@@ -43,28 +43,59 @@ jest.mock('@utils/config', () => ({
 const mockedHasReliableWebsocket = jest.mocked(hasReliableWebsocket);
 const mockedGetOrCreateWebSocketClient = jest.mocked(getOrCreateWebSocketClient);
 
+// Helper functions to handle Jest timer issues with async code
+// These are needed because Jest's fake timers don't play well with Promise-based code.
+// The combination of fake timers for time advancement + real timers for nextTick
+// allows us to properly test async timing behavior.
+const enableFakeTimers = () => {
+    jest.useFakeTimers({doNotFake: ['nextTick']});
+};
+
+const disableFakeTimers = () => {
+    jest.useRealTimers();
+};
+
+const advanceTimers = async (ms: number) => {
+    jest.advanceTimersByTime(ms);
+    await new Promise(process.nextTick);
+};
```

Comment:
I wonder if we should move this functions to the test_helper or maybe even create it's own file under the `test` folder, say .. timers_helper.ts ?

Reply:
I can make this change 👍